app/controllers/api/deploys_controller.rb

```diff
@@ -1,16 +1,43 @@
 # frozen_string_literal: true
 class Api::DeploysController < Api::BaseController
-  include CurrentProject
-
   skip_before_action :require_project, only: [:active_count]
 
   def index
-    scope = current_project.try(:deploys) || Deploy
-    @deploys = paginate(params[:ids].present? ? [scope.find(params[:ids])].flatten : scope)
-    render json: @deploys
+    render json: paginate(deploy_scope)
   end
 
   def active_count
     render json: Deploy.active.count
   end
+
+  protected
+
+  def filter
+    if filter_param = params.fetch(:filter, false)
+      filter_param if ["pending", "running", "canceled", "failed", "errored", "succeeded"].include?(filter_param)
+    else
+      false
+    end
+  end
+
+  def search_params
+    if project_id = params.fetch(:project_id, false)
+      stage_ids = Project.find(project_id).stages.pluck(:id)
+      search_params = {jobs: {}, deploys: {stage_id: stage_ids}}
+    elsif stage_id = params.fetch(:stage_id, false)
```

Comment:
sure. In the end it's moot, there are three routes that hit this method:

api/stages/:stage_id/deploys
api/projects/:project_id/deploys
api/deploys